api/v1alpha1/etcdcluster_types.go

```diff
@@ -130,6 +130,82 @@ type ProviderCertManagerConfig struct {
 type EtcdClusterStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
+
+	// ObservedGeneration is the most recent generation observed for this EtcdCluster by the controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// CurrentReplicas is the number of etcd pods managed by the StatefulSet for this cluster.
+	// This reflects the .spec.replicas of the underlying StatefulSet.
+	// +optional
+	CurrentReplicas int32 `json:"currentReplicas,omitempty"`
+
+	// ReadyReplicas is the number of etcd pods managed by the StatefulSet that are currently ready.
+	// This reflects the .status.readyReplicas of the underlying StatefulSet.
+	// +optional
+	ReadyReplicas int32 `json:"readyReplicas,omitempty"`
+
+	// MemberCount is the number of members currently registered in the etcd cluster,
+	// as reported by the etcd 'member list' API. This may differ from CurrentReplicas
+	// during scaling operations or if members are added/removed outside the operator's direct control.
+	// +optional
+	MemberCount int32 `json:"memberCount,omitempty"`
+
+	// CurrentVersion is the observed etcd version of the cluster.
+	// This is typically derived from the version of the healthy leader or a consensus among healthy members.
+	// +optional
+	CurrentVersion string `json:"currentVersion,omitempty"`
+
+	// LeaderId is the hex-encoded ID of the current etcd cluster leader, if one exists and is known.
+	// +optional
+	LeaderId string `json:"leaderId,omitempty"`
```

Comment:
Can we rename this to `LeaderID` to match the Go standard?

```suggestion
	// LeaderID is the hex-encoded ID of the current etcd cluster leader, if one exists and is known.
	// +optional
	LeaderID string `json:"leaderID,omitempty"`
```

Reply:
Yeah, it also matches what's in `etcdutils.go`. Thanks for the suggestion! Applied.